docs/source/theory.rst

```diff
@@ -4,5 +4,337 @@
 Theory
 ======
 
+--------------
+Bulk physics 
+--------------
 
-WIP
\ No newline at end of file
+H transport
+^^^^^^^^^^^
+
+The model developed by McNabb & Foster [:ref:`1 <McNabb and Foster>`] is used to model hydrogen transport in materials in FESTIM. The principle is to separate mobile hydrogen :math:`c_\mathrm{m}` and trapped hydrogen :math:`c_\mathrm{t}`. The diffusion of mobile particles is governed by Fick’s law of diffusion where the hydrogen flux is
+
+.. math::
+    :label: eq_difflux
+    
+    J = -D \nabla c_\mathrm{m}
+
+where :math:`D=D(T)` is the diffusivity. Each trap :math:`i` is associated with a 
+trapping and a detrapping rate :math:`k_i` and :math:`p_i`, respectively, as well 
+as a trap density :math:`n_i`.
+
+The temporal evolution of :math:`c_\mathrm{m}` and :math:`c_{\mathrm{t}, i}` are then 
+given by:
+
+.. math::
+    :label: eq_mobile_conc
+
+    \frac {\partial c_{\mathrm{m}, i}} { \partial t} = \nabla \cdot (D\nabla c_\mathrm{m}) - \sum_i \frac{\partial c_{\mathrm{t},i}}{\partial t} + \sum_j S_j
```

Comment:
@RemDelaporteMathurin, did you mean that FESTIM is able to deal with multiple mobile species? Or index $i$ was intended to $c_{t,i}$ below?

Reply:
Sorry yes, c_t,i